vernac/vernacentries.ml

```diff
@@ -534,17 +534,14 @@ let vernac_assumption ~atts discharge kind l nl =
   if not status then Feedback.feedback Feedback.AddedAxiom
 
 let should_treat_as_cumulative cum poly =
-  if poly then
-    match cum with
-    | GlobalCumulativity | LocalCumulativity -> true
-    | GlobalNonCumulativity | LocalNonCumulativity -> false
-  else
-    match cum with
-    | GlobalCumulativity | GlobalNonCumulativity -> false
-    | LocalCumulativity -> 
-      user_err Pp.(str "The Cumulative prefix can only be used in a polymorphic context.")
-    | LocalNonCumulativity ->
-      user_err Pp.(str "The NonCumulative prefix can only be used in a polymorphic context.")
+  match cum with
+  | Some true ->
+    if poly then true
+    else user_err Pp.(str "The Cumulative prefix can only be used in a polymorphic context.")
+  | Some false ->
+    if poly then false
+    else user_err Pp.(str "The NonCumulative prefix can only be used in a polymorphic context.")
+  | None -> poly && Flags.is_polymorphic_inductive_cumulativity ()
```

Comment:
Is this the (only) "relevant" line which makes the PR fixes the issue? (Or do I miss the point?)

If so, let me hijack (again) a PR. I understood that there was some kind of consensus, as a documentation, towards using more types with expliclty named constructions rather than booleans, as well as using labels to document booleans governing optional behaviors (at least, this was what @maximedenes was recommending at some time, and this sounds relevant to me). So, what is the general opinion about maybe trying to preserve the names `LocalCumulativity`, etc., for easier understanding and reasoning about the code?

Reply:
> Is this the (only) "relevant" line which makes the PR fixes the issue? (Or do I miss the point?)

Accessing Flags at parse time is not ok. The change to `bool option` is orthogonal.

Reply:
I wouldn't say that it's orthogonal, you have to put in something when there's no local flag.